Enable torchinductor opinfo tests for cpp_wrapper

benjaminglass1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141699
* #141371
* #142286

Additionally, do some minor bugfixes that didn't merit a PR of their
own.

Note: I've sharded the cpp_wrapper CI checks into 2 shards.  More would
be helpful given the compilation slowdown associated with cpp_wrapper,
but sharding happens on a file-by-file basis rather than test-by-test.
We should definitely reconsider that in the future.

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update on "Enable torchinductor opinfo tests for cpp_wrapper"


Additionally, do some minor bugfixes that didn't merit a PR of their
own.

Note: I've sharded the cpp_wrapper CI checks into 2 shards.  More would
be helpful given the

[ghstack-poisoned]

benjaminglass1 committed:
Update on "Enable torchinductor opinfo tests for cpp_wrapper"


Additionally, do some minor bugfixes that didn't merit a PR of their
own.

Note: I've sharded the cpp_wrapper CI checks into 2 shards.  More would
be helpful given the compilation slowdown associated with cpp_wrapper,
but sharding happens on a file-by-file basis rather than test-by-test.
We should definitely reconsider that in the future.

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

huydhn commented:
> @huydhn , this PR hits a problem that `test_torchinductor_opinfo` is not sharded and runs into timeout. Meanwhile, I have seen `test_torchinductor` is properly sharded. Do you know what changes we need in order to make test_torchinductor_opinfo conform with the existing sharding mechanism?

This change looks correct to me, adding the shard number is the way to go, i.e.

```
  python test/run_test.py \
    --include inductor/test_torchinductor inductor/test_torchinductor_opinfo \
    --shard "$1" "$NUM_TEST_SHARDS" \
    --verbose
```

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 commented:
@huydhn Do you have any insight into why the sharding seems to work differently in the pipeline than locally? Manually using `test/run_test.py` with 2 shards on my local machine appears to run `test_torchinductor` in one shard and `test_torchinductor_opinfo` in a different shard, but that's not what happens here.

benjaminglass1 committed:
Update on "Enable torchinductor opinfo tests for cpp_wrapper"


Additionally, do some minor bugfixes that didn't merit a PR of their
own.

Note: I've sharded the cpp_wrapper CI checks into 2 shards.  More would
be helpful given the compilation slowdown associated with cpp_wrapper,
but sharding happens on a file-by-file basis rather than test-by-test.
We should definitely reconsider that in the future.

[ghstack-poisoned]

benjaminglass1 committed:
Update on "Enable torchinductor opinfo tests for cpp_wrapper"


Additionally, do some minor bugfixes that didn't merit a PR of their
own.

Note: I've sharded the cpp_wrapper CI checks into 2 shards.  More would
be helpful given the compilation slowdown associated with cpp_wrapper,
but sharding happens on a file-by-file basis rather than test-by-test.
We should definitely reconsider that in the future.

[ghstack-poisoned]

huydhn commented:
> @huydhn Do you have any insight into why the sharding seems to work differently in the pipeline than locally? Manually using `test/run_test.py` with 2 shards on my local machine appears to run `test_torchinductor` in one shard and `test_torchinductor_opinfo` in a different shard, but that's not what happens here.

Sharding is meant to be used in CI only because it relies on the test time from https://github.com/pytorch/test-infra/blob/generated-stats/stats/test-times.json to balance the shard.  When you run it locally, this information is not available, so I think the script just round robins the the 2 tests into these 2 shards.

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]

benjaminglass1 committed:
Update

[ghstack-poisoned]
